pci/src/vfio.rs

```diff
@@ -1912,6 +1956,13 @@ impl<M: GuestAddressSpace + Sync + Send> ExternalDmaMapping for VfioDmaMapping<M
         let guest_addr = GuestAddress(gpa);
         let user_addr = if mem.check_range(guest_addr, size as usize) {
             mem.get_host_address(guest_addr).unwrap() as u64
+        } else if self.mmio_regions.lock().unwrap().check_range(gpa, size) {
+            self.mmio_regions
+                .lock()
+                .unwrap()
+                .find_user_address(gpa)
+                .ok_or("unable to locate user addr for mmio region")
+                .unwrap()
```

Comment:
Nit: Instead of using `unwrap()`, it is better to return an Error with the failing `gpa`, so that the Cloud Hypervisor process won't be aborted yet providing some information for debugging. You can take the following `else` statement as a reference.

Reply:
Noted - I've removed the `unwrap()` usage and returned a more helpful message for the error case (for both the guest_memory and mmio checks)